Implement popup windows

TyGriffin115 opened the pull request:
Implemented Popup Warnings for every Delete Button in the front-end. The delete functionality for the scenario delete buttons on the logistics page and dashboard page have yet to be implemented. We should wait until after the integration is complete for the logistics and dashboard pages. 

![Gif1](https://user-images.githubusercontent.com/54215233/99490074-1825c280-2937-11eb-945e-c7c8518eeab2.gif)

![Gif12](https://user-images.githubusercontent.com/54215233/99490376-bade4100-2937-11eb-98d6-e44d0b596cc7.gif)

TyGriffin115 committed:
Added Popup windows for Delete buttons on sidebar in editor

TyGriffin115 committed:
Added Warnings for Every Delete Button

Added Popup warnings for every delete button in the front end.

Only two delete buttons do not have functionailty and need to be finished. These are the ones that delete the entire scenario.

eneadodi commented on Ghost_in_the_Shell/client_folder_react_app/ethisim/src/components/ConfigurationSideBarComponents/NavSideBarNode.js:
```diff
@@ -77,15 +74,7 @@ export default function NavSideBarNode(props) {
-                            <DeleteForeverIcon />
-                        </Button>
-                    </Grid>
+                    <DeletePopupSidebar id={id} deleteByID={deleteByID} />
```
Nice job working with the current implementation!

TyGriffin115 replied:
Fixed! There are only two files now, one for a generic delete warning and the other is a warning for deleting an entire scenario and all its data.

eneadodi reviewed (dismissed):
There probably was a way to implement the popup so you didn't have to make a different component for each page type. Nevertheless it seems to work out, just was more work for you. I approve!

TyGriffin115 committed:
Used a single dialog component for delete Warnings

There are now only two files, a generic delete warning and another warning for deleting a whole scenerio, this is more efficient than the last implementation.

TyGriffin115 committed:
Merge branch 'master' into Implement-Popup-windows

TyGriffin115 committed:
Small typo Fix

eneadodi reviewed (dismissed):
Looks good! Well done.

TyGriffin115 committed:
Quick Fix for merge

TyGriffin115 committed:
Merge branch 'master' into Implement-Popup-windows

TyGriffin115 committed:
More Quick Fixes after merge

Enoch-Hsiao reviewed (approved):
LGTM
